snuba/snuba_migrations/replays/0019_add_materialization.py

```diff
@@ -0,0 +1,192 @@
+from typing import Iterator, List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Column,
+    DateTime,
+    IPv4,
+    IPv6,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(forward_iter())
+
+    def backwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(backward_iter())
+
+
+def forward_iter() -> Iterator[operations.SqlOperation]:
+    yield operations.CreateTable(
+        storage_set=StorageSetKey.REPLAYS,
+        table_name="replays_aggregated_local",
+        columns=columns,
+        engine=table_engines.AggregatingMergeTree(
+            storage_set=StorageSetKey.REPLAYS,
+            order_by="(project_id, toStartOfHour(to_hour_timestamp), replay_id)",
+            partition_by="(retention_days, toMonday(to_hour_timestamp))",
+            settings={"index_granularity": "8192"},
+            ttl="to_hour_timestamp + toIntervalDay(retention_days)",
+        ),
+        target=operations.OperationTarget.LOCAL,
+    )
+
+    yield operations.AddIndex(
```

Comment:
Why do you need this index if `replay_id` is already in the primary key?

Reply:
@evanh Right because we can binary search for the replay-id.  I can remove it.